examples/hosts/app-integration/corruption-recovery/src/app.tsx

```diff
@@ -0,0 +1,150 @@
+/*!
+ * Copyright (c) Microsoft Corporation and contributors. All rights reserved.
+ * Licensed under the MIT License.
+ */
+import React from "react";
+import ReactDOM from "react-dom";
+
+import { AzureClient } from "@fluidframework/azure-client";
+
+import { IFluidContainer, SharedMap } from "fluid-framework";
+import { DataController } from "./dataController";
+import { RecoveryAgent } from "./recoveryAgent";
+import { connectionConfig } from "./azureConfig";
+
+import { AppView } from "./appView";
+
+// Define the schema of our Container.
+const containerSchema = {
+    initialObjects: {
+        dataMap: SharedMap,
+        recoveryMap: SharedMap,
+    },
+};
+
+async function initializeData(container: IFluidContainer): Promise<void> {
+    const sharedMap = container.initialObjects.dataMap as SharedMap;
+    await Promise.all([DataController.initializeModel(sharedMap)]);
+}
+
+async function createRecoveryAgent(
+    container: IFluidContainer,
+    containerId: string,
+): Promise<void> {
+    const rollContainerId = await RecoveryAgent.createRecoveryAgent(
+        containerId,
+    );
+    const recMap = container.initialObjects.recoveryMap as SharedMap;
+    recMap.set("recoverContainerId", rollContainerId);
+}
+
+function cleanRecoveryAgent(container: IFluidContainer): void {
+    const recMap = container.initialObjects.recoveryMap as SharedMap;
+    recMap.set("recoverContainerId", "");
+}
+
+async function getRecoveryAgent(
+    container: IFluidContainer,
+): Promise<RecoveryAgent | undefined> {
+    const recMap = container.initialObjects.recoveryMap as SharedMap;
+    let recoveryContainerId = recMap.get<string>("recoverContainerId");
+
+    if (!(recoveryContainerId ?? "")) {
+        await new Promise<void>((resolve) =>
+            recMap.once("valueChanged", () => resolve()),
+        );
+        recoveryContainerId = recMap.get("recoverContainerId");
```

Comment:
It's not very clear what is the intention here. Do you expect empty string here? If so, assert would be nice after it.
If you expect anything, then I'd expect it to be a loop (like 52), not an if